### PR TITLE
docs(pack-up): add readme and contributor docs

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* adds README
* adds contributor docs

### Why is it needed?

* Good documentation makes people happy

### Related issue(s)/PR(s)

* resolves CONTENT-1905

### CODE REVIEW

File: docs/docs/docs/05-utils/pack-up/00-intro.mdx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+title: Introduction
+tags:
+  - pack-up
+  - bundling
+  - packages
+---
+
+## What is pack-up?
+
+Pack-Up is primarily a CLI interface to develop & build CJS/ESM interopable packages. It's built on top of [vite](https://vitejs.dev/),
```
Comment (Marc-Roig) [inline docs/docs/docs/05-utils/pack-up/00-intro.mdx:11]: comming from a lack of understanding of bundling and transpiling, what interopable packages mean? packages that both can be imported from cjs and esm projects?

Comment (joshuaellis) [reply docs/docs/docs/05-utils/pack-up/00-intro.mdx:11]: yes!

Comment (joshuaellis) [reply docs/docs/docs/05-utils/pack-up/00-intro.mdx:11]: I could add a link to the meaning?

Comment (Marc-Roig) [reply docs/docs/docs/05-utils/pack-up/00-intro.mdx:11]: yes! that would help!

File: docs/docs/docs/05-utils/pack-up/00-intro.mdx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+to build other tools on top of it.
+
+It's designed to have as little dependencies as possible making it easier to react to changes in the ecosystem as and when we need too.
+The configuration format at present is limited and is expected to grow with real-world requirements.
+
+## Why do we need it?
+
+Creating packages that work in both CJS and ESM environments is a pain. Strapi also has it's own plugin system that realistically
+would benefit from having that code bundled for both the client and server packages of said plugins. Initially we wanted to keep
+this inside the `@strapi/strapi` package but realised soon after that this package was required to be more low level than the former.
```
Comment (Marc-Roig) [inline docs/docs/docs/05-utils/pack-up/00-intro.mdx:22]: I would have an aditional question, Who will be using this?

At which point of strapi development would you use this? 
Will this be used automatically when publishing and building strapi?

Comment (joshuaellis) [reply docs/docs/docs/05-utils/pack-up/00-intro.mdx]: updated ✅

File: docs/docs/docs/05-utils/pack-up/00-intro.mdx (hunk 1 above)
Comment (joshuaellis) [reply docs/docs/docs/05-utils/pack-up/00-intro.mdx]: updated ✅

**Review by Marc-Roig — APPROVED** [review]

**Pull request merged**